Fix missing bio_new

noellabo opened the pull request:
- Resolve errors related to 'bio_new' occurring in crystal 0.27.0

Fixes #2

byfoz commented:
@noellabo huge thanks for this! 🙌🏻

Just want to check before merging in:
> I was worried about errors, so I wrote a fix.
>
> It will work with OpenSSL 1.0.2, but because it is not solving the essential problem in 1.1 and later it should not work yet.

Can you elaborate on this? ...
>but because it is not solving the essential problem in 1.1 and later it should not work yet

I'm not sure what the essential problem is in this context - it would be great if we can support more than OpenSSL v1.0.2

noellabo commented:
Fixed the problem that the spec did not pass.

- Since the key length of RSA specified in spec is too short in OpenSSL 1.1.1, we changed the minimum value to 512. (Segmentation fault and some failures occurred.)
- In PEM decode including blank lines, returning 0 to BIO_gets in OpenSSL 1.1.1 seems to be judged as a termination. As a workground, when reading only 1 byte, 1 is returned.

I think that the judgment of the blank line of gets of OpenSSL 1.1.1 is around here.
https://github.com/openssl/openssl/commit/98e553d2ce31e2179be68d6a60b5bec765cd9768#diff-77e13ca6978d539e6df17cdb50830082L387

If someone can make a better correction, please solve it.

CImrie commented on src/openssl_ext/bio.cr:
```diff
@@ -16,21 +16,28 @@ struct OpenSSL::GETS_BIO
-    @bio = LibCrypto.bio_new(pointerof(GETS_BIO))
+    @bio = LibCrypto.BIO_new(GETS_BIO)
 
     # We need to store a reference to the box because it's
```
```suggestion

```

CImrie commented on src/openssl_ext/bio.cr:
```diff
@@ -16,21 +16,28 @@ struct OpenSSL::GETS_BIO
+    @bio = LibCrypto.BIO_new(GETS_BIO)
 
     # We need to store a reference to the box because it's
     # stored in `@bio.value.ptr`, but that lives in C-land,
```
```suggestion

```

CImrie commented on src/openssl_ext/bio.cr:
```diff
@@ -16,21 +16,28 @@ struct OpenSSL::GETS_BIO
 
     # We need to store a reference to the box because it's
     # stored in `@bio.value.ptr`, but that lives in C-land,
     # not in Crystal-land.
```
```suggestion

```
